### PR TITLE
Updates OctoPack.nuspec to use best-practice layout

### PR DESCRIPTION
**Pull request by fourpastmidnight**
This commit updates the OctoPack.nuspec file to use a more best-practice
layout, as can be seen at [Nate McMaster's blog (MS)](https://natemcmaster.com/blog/2017/11/11/build-tools-in-nuget).

This commit:

* Updates the `build.cake` script to remove the `.\build\contents\<tfm>`
folder creation and instead creates `.\build\lib\<tfm>` folders, as this
is currently the best-practice for creating NuGet packaged build tools
such that they report to Visual Studio the target frameworks for which
the tools are supported.

* Updates the NuSpec file accordingly, using a `lib` folder in place of
the `content` folder. Also, this NuGet package can only be installed
with a NuGet client having a minimum  version of 2.5 or greater, so this
was made explicit by adding the `minClientVersion` attribute to the
Nuspec's `<metadata />` element.

It's really hard to write a "portable end-to-end test" that the package
installs/uninstalls using both `packages.config`-style and
`<PackageRefenence />`-style installation/uninstallation methods, since
I'm not likely to have a development NuGet feed setup in just the same
way as someone else. But I did test these changes locally for both
styles of installing OctoPack, building the test project with
`/t:Build /p:Configuration=Release /p:RunOctoPack=true` and verifying
the OctoPacked package contents, and then uninstalling OctoPack.

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -94,10 +94,10 @@ Task("Package")
     .IsDependentOn("Test")
     .Does(() => {
         CreateDirectory(buildDir);
-        CreateDirectory(Path.Combine(buildDir, "content"));
-        CreateDirectory(Path.Combine(buildDir, "content", "net35"));
-        CreateDirectory(Path.Combine(buildDir, "content", "net40"));
-        CreateDirectory(Path.Combine(buildDir, "content", "netcore45"));
+        CreateDirectory(Path.Combine(buildDir, "lib"));
```
Comment (tothegills) [inline build.cake:97]: Looking at the [multi-targeting documentation](https://docs.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks#framework-version-folder-structure), I'm wondering if the `lib` folder has the same effect as the old `content` folder? Wonder if we need these at all.

File: source/OctoPack.nuspec [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
-  <metadata>
+  <metadata minClientVersion="2.5">
```
Comment (tothegills) [inline source/OctoPack.nuspec:3]: This was introduced in [NuGet 2.5 in 2013] (https://devblogs.microsoft.com/nuget/a-quick-note-about-minclientversion-feature/), I think we're safe.

File: build.cake (hunk 1 above)
Comment (fourpastmidnight) [reply build.cake:97]: In this case, yes, the `lib` folder replaces what the `content` folder provided at the time. I'm not sure if you can just do away with the `lib` folder or not. But this is saying that this package can be installed into projects targeting .NET 3.5, 4.0, and Windows 8 "store" apps.

File: source/OctoPack.nuspec (hunk 2 above)
Comment (fourpastmidnight) [reply source/OctoPack.nuspec:3]: Well, I was here, I came across the information, and so I added it. No harm no foul, really. Hopefully, no one's using Visual Studio 2010 or 2012 trying to build OctoPack at this time! 😄  The more important change is the one above to be a bit more in line with "best practices".

**Pull request merged**